app/services/movie_data_service.rb

```diff
@@ -250,5 +249,35 @@ def get_common_movies_between_multiple_actors(actor_names: nil, paginate_actor_n
         total_pages: movie_response[:total_pages]
       )
     end
+
+    def get_movie_streaming_service_providers(movie)
+      results = Tmdb::Client.request(:streaming_service_providers, movie_id: movie.tmdb_id).dig(:results, :US)
+      preferred_providers = [
+        { name: 'Netflix', pay_model: 'try', url: "http://www.netflix.com/search/#{movie.title}"},
+        { name: 'Amazon Prime Video', pay_model: 'try', url: "https://www.amazon.com/s?k=#{movie.title.gsub(' ', '+')}&i=instant-video"},
+        { name: 'Amazon Video', pay_model: 'try', url: "https://www.amazon.com/s?k=#{movie.title.gsub(' ', '+')}&i=instant-video"},
+        { name: 'YouTube', pay_model: 'try', url: "https://www.youtube.com/results?search_query=#{movie.title}+full+movie #{movie.release_date.stamp('2001') if movie.release_date.present?}"},
+        { name: 'Vudu', pay_model: 'try', url: "https://www.vudu.com/content/movies/search?searchString=#{movie.title}"},
+      ]
+      return preferred_providers if results.nil?
+
+      free_provider_names = results[:flatrate]&.map { |result| result[:provider_name] } || []
+      rent_provider_names = results[:rent]&.map { |result| result[:provider_name] } || []
+      buy_provider_names = results[:buy]&.map { |result| result[:provider_name] } || []
+
+      preferred_providers.map do |provider|
```

Comment:
The order of this conditional block is important. It ensures that we're displaying the least-expensive option for each provider. If a provider does not appear in any of the payment model results, it is assigned a "try", which translates in the view as "go look it up, we don't know".